### PR TITLE
Launch cleanup

### PR DESCRIPTION
**Pull request by peter-edge**
btrfs is 3.17 on coreos beta channel.

### CODE REVIEW

**Comment by peter-edge** [comment]
I got through most of the scripts, still a few left. What mostly inspired this was difficulty getting a dev envinronment up and running, and wanting more clarity as an external user as to what was happening - I think this a decent start on that :) I would run through the commands yourself, however, to make sure I did not have any typos. Ie I do have typos and am still not having the easiest time running all of them :)

**Comment by jdoliner** [comment]
Many thanks for the pull! 
Would you mind signing our [CLA](https://www.clahub.com/agreements/pachyderm/pfs)?
I'l try to take a look at this first thing on Monday.
Send me your tshirt size and mailing address at jd@pachyderm.io for some swag.

**Comment by peter-edge** [comment]
Hmm, I've tried both Chrome and Safari and CLAHub and both are returning an error. Can you see on your side if it actually went through?

image

**Comment by JoeyZwicker** [comment]
Hmm, https://www.clahub.com/agreements/pachyderm/pfs works just fine for
me. We'll figure it out.

**Comment by peter-edge** [comment]
It's alright. The PR isn't really complete anyways I'd say, if you're
comfortable with the direction of it I'll finish it and we will figure this
out in the mean time.

**Comment by jdoliner** [comment]
Yeah turns out CLAHub is now defunct. We'll figure something else out.

**Comment by peter-edge** [comment]
Lol

**Comment by jdoliner** [comment]
Sorry for the delay.

First off I really like the direction of this PR. Makes our scripts much more sane. Big :+1: on this.

Second, round 2 of our CLA is ready, this time I'm 99% sure that it works :D https://pachyderm.wufoo.com/forms/pachyderm-contributor-license-agreement/

**Comment by peter-edge** [comment]
lol got it. you should update the link in CONTRIBUTING.md :)

this is generally at the state where I'd want you to verify stuff works, also covers https://github.com/pachyderm/pfs/issues/70

also see https://github.com/pachyderm/pfs/issues/81

overall, ya this is a start, but definitely would want to do a PR after this that moves a lot of stuff to makefiles, and moves the -wrapper scripts somewhere else (to denote they are for use interally only).

**Comment by peter-edge** [comment]
note this also takes care of https://github.com/pachyderm/pfs/pull/67 but in a very different/more complicated way

File: scripts/lib/common.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  local version="${2}"
+  local major="${3}"
+  local minor="${4}"
+
+  if ! which "${executable}" > /dev/null; then
+    echo "error: ${executable} not installed" >&2
+    return 1
+  fi
+
+  local regex="([0-9]+)\.([0-9]+)\.([0-9]+)"
```
Comment (bitwiseman) [inline scripts/lib/common.sh]: Support version strings that don't contain micro:
`([0-9]+)\.([0-9]+)(\.([0-9]+))?`

File: scripts/lib/common.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  if [ "$(echo "${version}" | sed -E "s/${regex}/\1/")" -ne "${major}" ]; then
+    echo "error: ${executable} version ${version} must have major version ${major}" >&2
```
Comment (bitwiseman) [inline scripts/lib/common.sh]: `error: Found ${executable} version ${version}. Major version "${major}" required.`

File: scripts/lib/common.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    return 1
+  fi
+  if [ "$(echo "${version}" | sed -E "s/${regex}/\2/")" -lt "${minor}" ]; then
```
Comment (bitwiseman) [inline scripts/lib/common.sh]: See #93 maybe merge these two if's into one.

**Comment by jdoliner** [comment]
Hmm, I'm getting an issue with how the scripts source each other :(

```
$ scripts/clean
scripts/clean: 10: scripts/clean: source: not found
```

This is run from the root of the source tree. Is this a system specific issue?

**Comment by bitwiseman** [comment]
@peter-edge,
If you fetch my fork, I've done the merge and added a commit for the comments I made. 
https://github.com/bitwiseman/pfs/commit/691da176be93a9d906b862cea117e1671cfa1cb0

**Comment by bitwiseman** [comment]
@jdoliner - what shell/version are you running? That error message seems to say that the `source` command wasn't found...

**Comment by peter-edge** [comment]
@bitwiseman can you create PRs for those two changes against peter-edge/pfs, so i can merge them in?  @jdoliner: using `.` instead of `source` should solve this problem, i'm used to doing things in a bashy way, which is a bad habit for docker world

**Comment by jdoliner** [comment]
I'm on zsh 5.0.5. Using `.` makes sense :D

**Comment by peter-edge** [comment]
dasdvkjasdvkbasdvkbjasdv well at least we'll know all our scripts will be compatible with both bash and zsh hahaha :)

**Comment by bitwiseman** [comment]
@peter-edge  - https://github.com/peter-edge/pfs/pull/1

**Comment by peter-edge** [comment]
merged :)

**Comment by bitwiseman** [comment]
@peter-edge - do you think you could compress (rebase -i) your changes into a meaningful set of commits?   "more cleanup" is not a great commit history.

**Comment by peter-edge** [comment]
@bitwiseman I'm having a lot of trouble with this, this is not a technique I have used before, I may just patch the whole thing into a new PR, what do you think?

**Comment by bitwiseman** [comment]
You don't have to go to a whole new PR, you can just force push into this same one... 
This is also just my suggestion, so ... You're not required to do it. :smile:

**Pull request merged**